pytorch_lightning/strategies/collaborative.py

```diff
@@ -0,0 +1,509 @@
+    """
+
+    def __init__(self, optimizer: "hivemind.Optimizer", scheduler: _LRScheduler) -> None:
+        # copy most of the `Scheduler` methods into this instance. `__del__` is skipped in case the scheduler has
+        # implemented custom logic which we would not want to call on destruction of the `HiveMindScheduler`
+        self.__dict__ = {k: v for k, v in scheduler.__dict__.items() if k not in ("step", "__del__")}
+
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.current_step = -1
+
+    def step(self, epoch: Optional[int] = None) -> None:
+        while self.current_step < self.optimizer.local_epoch:
+            self.scheduler.step(epoch=epoch)
+            self.current_step += 1
+
+    def load_state_dict(self, state_dict: Dict) -> None:
+        self.scheduler.load_state_dict(state_dict)
+
+    def state_dict(self) -> Dict:
+        return self.scheduler.state_dict()
+
+
+class DHTManager:
+    ENDPOINT_ENV: str = "PL_ENDPOINT"
+    PEER_ENDPOINT_ENV: str = "PL_PEER_ENDPOINT"
+    INITIAL_PEERS_ENV: str = "PL_INITIAL_PEERS"
+    HOST_ENV: str = "PL_HOST"
+    PORT_ENV: str = "PL_PORT"
+    DEFAULT_HOST: str = "0.0.0.0"
+    DEFAULT_PORT: int = 1440
+
+    def __init__(
+        self,
+        host_maddrs: Optional[List],
+        initial_peers: Optional[Union[str, List]],
+        persistent: bool,
+        endpoint: Optional[bool],
+        peer_endpoint: Optional[str],
+        host: Optional[str],
+        port: Optional[int],
+        retry_endpoint_attempts: int = 5,
+        retry_endpoint_sleep_duration: int = 5,
+    ) -> None:
+        """Manages the `hivemind.DHT` connection and provides a side-car endpoint server for initial peer access.
+
+        Arguments:
+
+            host_maddrs: List of multi-addrs to create visible peers for other processes.
+                `https://learning-at-home.readthedocs.io/en/latest/user/dht.html#running-across-the-internet`
+
+            initial_peers: If connecting to a running process, a list of initial peers needs to be passed in.
+                This can also be set via the env variable `INITIAL_PEERS`.
+
+            endpoint: Enable if a side-car endpoint server is required on the process to server initial peers.
+                This is useful when using some form of orchestration such as torchelastic.
+
+            peer_endpoint: The endpoint to request initial peers from.
+
+            persistent: When using an endpoint, this controls whether other processes that are not the endpoint
+                server log/checkpoint. If `persistent` is True, we do not log/checkpoint from other processes.
+
+            host: When creating the endpoint, the host IP to use.
+
+            port: When creating the endpoint, the host port to use.
+
+            retry_endpoint_attempts: When connecting to the `peer_endpoint`, how many time to retry before raising
```

Comment:
I think the `CollaborativeStrategy` docstring could paramref all these arguments or the other way around to avoid duplication